core/dnsserver/register.go

```diff
@@ -20,13 +20,14 @@ const serverType = "dns"
 // wise they potentially clash with other server types.
 func init() {
 	flag.StringVar(&Port, serverType+".port", DefaultPort, "Default port")
+	port := Port
 
 	caddy.RegisterServerType(serverType, caddy.ServerType{
 		Directives: func() []string { return Directives },
 		DefaultInput: func() caddy.Input {
 			return caddy.CaddyfileInput{
 				Filepath:       "Corefile",
-				Contents:       []byte(".:" + Port + " {\nwhoami\n}\n"),
+				Contents:       []byte(".:" + port + " {\nwhoami\n}\n"),
 				ServerTypeName: serverType,
 			}
```

Comment:
Same thing here : if we want it to be clean. then Setter/Getter around the port variable, with mutex. and a cpmment say it is useful because reload